src/typeinfo.rs

```diff
@@ -0,0 +1,1159 @@
+        ret
+    }
+}
+
+impl TypeDescriptionMethods for TypeDescription {
+    fn data(&self) -> &RpcMap {
+        &self.data
+    }
+    fn data_mut(&mut self) -> &mut RpcMap {
+        &mut self.data
+    }
+}
+
+impl TryFrom<RpcValue> for TypeDescription {
+    type Error = String;
+
+    fn try_from(value: RpcValue) -> Result<Self, Self::Error> {
+        let map: RpcMap = value.try_into()?;
+        Ok(Self::from_rpc_map(map))
+    }
+}
+
+impl TryFrom<&RpcValue> for TypeDescription {
+    type Error = String;
+
+    fn try_from(value: &RpcValue) -> Result<Self, Self::Error> {
+        value.clone().try_into()
+    }
+}
+
+// impl From<TypeDescription> for RpcValue {
+//     fn from(value: TypeDescription) -> Self {
+//         value.into_rpc_map().into()
+//     }
+// }
+
+#[derive(Clone, Default)]
+pub struct FieldDescription {
+    data: RpcMap,
+}
+
+impl FieldDescription {
+    pub fn from_rpc_map(map: RpcMap) -> Self {
+        Self { data: merge_tags(map) }
+    }
+
+    pub fn into_rpc_map(self) -> RpcMap {
+        let is_descr_empty = self.description().is_none_or(str::is_empty);
+        let mut ret = self.data;
+        if is_descr_empty {
+            ret.remove(KEY_DESCRIPTION);
+        }
+        ret
+    }
+}
+
+impl From<FieldDescription> for RpcValue {
+    fn from(value: FieldDescription) -> Self {
+        value.into_rpc_map().into()
+    }
+}
+
+impl TryFrom<RpcValue> for FieldDescription {
+    type Error = String;
+
+    fn try_from(value: RpcValue) -> Result<Self, Self::Error> {
+        value.try_into().map(Self::from_rpc_map)
+    }
+}
+
+impl TryFrom<&RpcValue> for FieldDescription {
+    type Error = String;
+
+    fn try_from(value: &RpcValue) -> Result<Self, Self::Error> {
+        value.to_owned().try_into()
+    }
+}
+
+impl TypeDescriptionMethods for FieldDescription {
+    fn data(&self) -> &RpcMap {
+        &self.data
+    }
+    fn data_mut(&mut self) -> &mut RpcMap {
+        &mut self.data
+    }
+}
+impl FieldDescriptionMethods for FieldDescription { }
+
+#[derive(Clone, Default)]
+pub struct PropertyDescription {
+    base: FieldDescription,
+}
+
+impl PropertyDescription {
+    // returns (Self, extra tags)
+    pub fn from_rpc_map(mut map: RpcMap) -> (Self, RpcMap) {
+        const KNOWN_TAGS: &[&str] = &[
+            KEY_DEVICE_TYPE,
+            // KEY_SUPER_DEVICE_TYPE,
+            KEY_NAME,
+            KEY_TYPE_NAME,
+            KEY_LABEL,
+            KEY_DESCRIPTION,
+            KEY_UNIT,
+            KEY_METHODS,
+            "autoload",
+            "autorefresh",
+            "monitored",
+            "monitorOptions",
+            KEY_SAMPLE_TYPE,
+            KEY_ALARM,
+            KEY_ALARM_LEVEL,
+        ];
+
+        let node_map = KNOWN_TAGS
+            .iter()
+            .filter_map(|tag| map.remove(*tag).map(|val| (tag.to_string(), val)))
+            .filter(|(tag, _val)| tag != KEY_DEVICE_TYPE)
+            .collect::<RpcMap>();
```

Comment:
https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.extract_if

Can use this when it becomes stable :>